Feature:  Suggest a Product Form

ksookdhew opened the pull request:
<img width="1728" alt="Screenshot 2024-08-08 at 11 56 14" src="https://github.com/user-attachments/assets/6e90779a-54af-485f-a2a6-6e9f495534eb">
<img width="375" alt="Screenshot 2024-08-08 at 11 56 31" src="https://github.com/user-attachments/assets/9e393cb4-d9e3-4291-81d9-647c0c0f6d44">

ksookdhew committed:
feat: created suggest a product form

ksookdhew committed:
Merge branch: 'dev' into feature/suggestProduct

# Conflicts:
#	package-lock.json

ksookdhew committed:
feat: Navigation to suggest a product

ksookdhew committed:
feat: functionality for form submit

ksookdhew committed:
feat: removed non-functional components

ksookdhew committed:
feat: Navigation to suggest a product on mobile view

AdeWet commented on src/features/suggestProduct/SuggestProduct.tsx:
```diff
@@ -0,0 +1,126 @@
+
+interface ProductSuggestion {
+    title: string;
+    price: number;
```
very cool! just a suggestion - asking a customer for a price as a suggestion for a product might not be ideal. Perhaps you can use price if someone wants to upload a product through your platform :D

ksookdhew replied:
Nice catch, that's very true actually. I think I'm going to leave it for now just for the purpose of the project to showcase a number input in the form. But definitely if this was a real-world application, that would be the way to go 😄

theadley commented on src/features/suggestProduct/SuggestProduct.tsx:
```diff
@@ -0,0 +1,126 @@
+import {getAllCategories} from "../../services/api.ts";
+import {useNavigate} from "react-router-dom";
+
+interface ProductSuggestion {
```
Might be worth separating the types outside of this so you can use the type in state mgt etc.

theadley commented on src/features/suggestProduct/SuggestProduct.tsx:
```diff
@@ -0,0 +1,126 @@
+            category: ''
+        },
+        validationSchema: Yup.object({
+            title: Yup.string().required('Title is required'),
```
min length maybe?

theadley commented on src/features/suggestProduct/SuggestProduct.tsx:
```diff
@@ -0,0 +1,126 @@
+            category: Yup.string().required('Category is required')
+        }),
+        onSubmit: (values) => {
+            console.log('Form values:', values);
```
Only for dev, remove for merge please

theadley commented on src/features/suggestProduct/SuggestProduct.tsx:
```diff
@@ -0,0 +1,126 @@
+        }),
+        onSubmit: (values) => {
+            console.log('Form values:', values);
+            alert(`${values.title} Product suggested successfully!`);
```
Alerts are fairly sucky user experiences because they force the user to stop what they're doing. If you can, rather use a toast or notification.

theadley commented on src/features/suggestProduct/SuggestProduct.tsx:
```diff
@@ -0,0 +1,126 @@
+                    />
+                    {formik.touched.price && formik.errors.price ? (
+                        <div className="text-red-500 text-sm">{formik.errors.price}</div>
+                    ) : null}
```
Change to short circuit instead of ternary if you have no "else"

ksookdhew committed:
fix: Implemented Pr comments

Removed console log, added minimum length requirement moved interface out and changed ternary to short circuit

ksookdhew committed:
feat: successful suggestion alert

Added a custom alert that is displayed when the user makes a successful product suggestion
